examples/graphbolt/disk_based_feature/node_classification.py

```diff
@@ -0,0 +1,518 @@
+    )
+    parser.add_argument(
+        "--mode",
+        default="pinned-pinned-cuda",
+        choices=[
+            "cpu-cpu-cpu",
+            "cpu-cpu-cuda",
+            "cpu-pinned-cuda",
+            "pinned-pinned-cuda",
+            "cuda-pinned-cuda",
+            "cuda-cuda-cuda",
+        ],
+        help="Graph storage - feature storage - Train device: 'cpu' for CPU and"
+        " RAM, 'pinned' for pinned memory in RAM, 'cuda' for GPU and GPU memory.",
+    )
+    parser.add_argument("--layer-dependency", action="store_true")
+    parser.add_argument("--batch-dependency", type=int, default=1)
+    parser.add_argument(
+        "--cpu-feature-cache-policy",
+        type=str,
+        default=None,
+        choices=["s3-fifo", "sieve", "lru", "clock"],
+        help="The cache policy for the CPU feature cache.",
+    )
+    parser.add_argument(
+        "--cpu-cache-size-in-gigabytes",
+        type=float,
+        default=0,
+        help="The capacity of the CPU cache, the number of features to store.",
+    )
+    parser.add_argument(
+        "--gpu-cache-size-in-gigabytes",
+        type=float,
+        default=0,
+        help="The capacity of the GPU cache, the number of features to store.",
+    )
+    parser.add_argument("--early-stopping-patience", type=int, default=25)
+    parser.add_argument(
+        "--sample-mode",
+        default="sample_neighbor",
+        choices=["sample_neighbor", "sample_layer_neighbor"],
+        help="The sampling function when doing layerwise sampling.",
+    )
+    parser.add_argument("--precision", type=str, default="high")
+    parser.add_argument("--enable-inference", action="store_true")
+    return parser.parse_args()
+
+
+def main():
+    torch.set_float32_matmul_precision(args.precision)
+    if not torch.cuda.is_available():
+        args.mode = "cpu-cpu-cpu"
+    print(f"Training in {args.mode} mode.")
+    args.graph_device, args.feature_device, args.device = args.mode.split("-")
+    args.overlap_feature_fetch = args.feature_device == "pinned"
+    # For now, only sample_layer_neighbor is faster with this option
+    args.overlap_graph_fetch = (
+        args.sample_mode == "sample_layer_neighbor"
+        and args.graph_device == "pinned"
+    )
+
+    # Load and preprocess dataset.
+    print("Loading data...")
+    disk_based_feature_keys = None
+    if args.cpu_cache_size_in_gigabytes > 0:
+        disk_based_feature_keys = [("node", None, "feat")]
+
+    dataset = gb.BuiltinDataset(args.dataset, root=args.root)
+    if disk_based_feature_keys is None:
+        disk_based_feature_keys = set()
+    for feature in dataset.yaml_data["feature_data"]:
+        feature_key = (feature["domain"], feature["type"], feature["name"])
+        # Set the in_memory setting to False without modifying YAML file.
+        if feature_key in disk_based_feature_keys:
+            feature["in_memory"] = False
+    dataset = dataset.load()
```

Comment:
And highlight here as well, explain what we are doing. When the user looks at this example after looking at examples/graphbolt/node_classification.py, they should see what is the difference.